inspirehep/modules/hal/config.py

```diff
@@ -20,10 +20,46 @@
 # granted to it by virtue of its status as an Intergovernmental Organization
 # or submit itself to any jurisdiction.
 
-"""HAL module configuration."""
+"""HAL Configuration."""
 
 from __future__ import absolute_import, division, print_function
 
 
-HAL_EXPORT_TEMPLATE = 'hal_tei.xml'
-"""Filename of the base TEI template in the template environment."""
+#
+# Configuration used when converting to TEI.
+#
+
+HAL_DOMAIN_MAPPING = {
+    'Accelerators': 'phys.phys.phys-acc-ph',
+    'Astrophysics': 'phys.astr',
+    'Computing': 'info',
+    'Data Analysis and Statistics': 'phys.phys.phys-data-an',
+    'Experiment-HEP': 'phys.hexp',
+    'Experiment-Nucl': 'phys.nexp',
+    'General Physics': 'phys.phys.phys-gen-ph',
+    'Gravitation and Cosmology': 'phys.grqc',
+    'Instrumentation': 'phys.phys.phys-ins-det',
+    'Lattice': 'phys.hlat',
+    'Math and Math Physics': 'math',
+    'Other': 'phys',
+    'Phenomenology-HEP': 'phys.hphe',
+    'Theory-HEP': 'phys.hthe',
+    'Theory-Nucl': 'phys.nucl',
+}
+"""Mapping used when converting from INSPIRE categories to HAL domains."""
+
+
+#
+# Configuration used when connecting to HAL.
+#
+
+HAL_COL_IRI = 'https://api-preprod.archives-ouvertes.fr/sword/hal'
+HAL_EDIT_IRI = 'https://api-preprod.archives-ouvertes.fr/sword/'
```

Comment:
Shall we rather specify here both `preprod` and `prod` IRIs and use a runtime flag (defaulting to `PRODUCTION`) in order to switch to the correct IRIs?

A bit like: https://github.com/inveniosoftware/invenio-oauthclient/blob/master/invenio_oauthclient/contrib/orcid.py

This would simplify deployment and operations, which should no longer be aware of exact IRIs (a knowledge that is in principle known only by who implemented HAL integration).